File: 2017_July/Day_Two/Nuts_and_Bolts/Fortran/mpi_hello.f90
PROGRAM MAIN
	Implicit None
    include "mpif.h"
	Character*8 :: rank_string, nproc_string
    Character(MPI_MAX_PROCESSOR_NAME) :: node_name
    Integer :: ierr, num_proc, my_rank, name_len

    Call MPI_INIT( ierr )
    Call MPI_Comm_size(MPI_COMM_WORLD, num_proc,ierr)
    Call MPI_Comm_rank(MPI_COMM_WORLD, my_rank,ierr)
    Call MPI_Get_processor_name(node_name, name_len,ierr)


    Write(rank_string,'(i8)')my_rank
    Write(nproc_string,'(i8)')num_proc

    If (my_rank .eq. 0) Then
        !Sometimes we might like for only a single thread to report certain information.
        !This avoids redundant output.
        Write(6,*)"  "//trim(adjustl(nproc_string))//" MPI processes are now active."
    Endif

    Write(6,*)"  Hello from node "//trim(node_name)//" rank "// &
        & trim(adjustl(rank_string))//" out of "//trim(adjustl(nproc_string))//" processors." 

    Call MPI_Finalize(ierr)
END PROGRAM MAIN

File: 2017_July/Day_Two/Nuts_and_Bolts/C++/mpi_hello.cpp
#include <stdio.h>
#include <mpi.h>
int main(int argc, char** argv) {
    int num_proc;  // The number of processes
    int my_rank; // the rank of this process (ranges from 0-nproc-1) 
    // The next string will hold the node's name.
    // Note that the value of MPI_MAX_PROCESSOR_NAME is defined by the MPI distribution
    char node_name[MPI_MAX_PROCESSOR_NAME]; // string to hold the node's name (e.g., shas0118)
    int name_len; // the number of characters in node_name

    MPI_Init(NULL, NULL);  
    MPI_Comm_size(MPI_COMM_WORLD, &num_proc);
    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
    MPI_Get_processor_name(node_name, &name_len);

    if (my_rank == 0)
    {
        //Sometimes we might like for only a single thread to report certain information.
        //This avoids redundant output.
        printf("  %d MPI Processes are now active.\n", num_proc);
    }

   
    printf("  Hello world from processor %s, rank %d out of %d processors\n",
         node_name, my_rank, num_proc);

    MPI_Finalize();
  
}

